numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if not issubclass(x.dtype.type, inexact):
+        x = x.astype(float)
```

Comment:
Not sure if I follow what you mean. I believe they are already handled. For instance, the following holds.

    >>> issubclass(np.complexfloating, np.inexact)
    True